src/components/portfolio/Details.tsx

```diff
@@ -24,430 +30,104 @@ const useStyles = makeStyles(({ palette }: Theme) =>
-  [1352156400000, 34.39],
-  [1352242800000, 33.47],
-  [1352329200000, 32.98],
-  [1352415600000, 32.9],
-  [1352674800000, 32.7],
-  [1352761200000, 32.54],
-  [1352847600000, 32.23],
-  [1352934000000, 32.64],
-  [1353020400000, 32.65],
-  [1353279600000, 32.92],
-  [1353366000000, 32.64],
-  [1353452400000, 32.84],
-  [1353625200000, 33.4],
-  [1353884400000, 33.3],
-  [1353970800000, 33.18],
-  [1354057200000, 33.88],
-  [1354143600000, 34.09],
-  [1354230000000, 34.61],
-  [1354489200000, 34.7],
-  [1354575600000, 35.3],
-  [1354662000000, 35.4],
-  [1354748400000, 35.14],
-  [1354834800000, 35.48],
-  [1355094000000, 35.75],
-  [1355180400000, 35.54],
-  [1355266800000, 35.96],
-  [1355353200000, 35.53],
-  [1355439600000, 37.56],
-  [1355698800000, 37.42],
-  [1355785200000, 37.49],
-  [1355871600000, 38.09],
-  [1355958000000, 37.87],
-  [1356044400000, 37.71],
-  [1356303600000, 37.53],
-  [1356476400000, 37.55],
-  [1356562800000, 37.3],
-  [1356649200000, 36.9],
-  [1356908400000, 37.68],
-  [1357081200000, 38.34],
-  [1357167600000, 37.75],
-  [1357254000000, 38.13],
-  [1357513200000, 37.94],
-  [1357599600000, 38.14],
-  [1357686000000, 38.66],
-  [1357772400000, 38.62],
-  [1357858800000, 38.09],
-  [1358118000000, 38.16],
-  [1358204400000, 38.15],
-  [1358290800000, 37.88],
-  [1358377200000, 37.73],
-  [1358463600000, 37.98],
-  [1358809200000, 37.95],
-  [1358895600000, 38.25],
-  [1358982000000, 38.1],
-  [1359068400000, 38.32],
-  [1359327600000, 38.24],
-  [1359414000000, 38.52],
-  [1359500400000, 37.94],
-  [1359586800000, 37.83],
-  [1359673200000, 38.34],
-  [1359932400000, 38.1],
-  [1360018800000, 38.51],
-  [1360105200000, 38.4],
-  [1360191600000, 38.07],
-  [1360278000000, 39.12],
-  [1360537200000, 38.64],
-  [1360623600000, 38.89],
-  [1360710000000, 38.81],
-  [1360796400000, 38.61],
-  [1360882800000, 38.63],
-  [1361228400000, 38.99],
-  [1361314800000, 38.77],
-  [1361401200000, 38.34],
-  [1361487600000, 38.55],
-  [1361746800000, 38.11],
-  [1361833200000, 38.59],
-  [1361919600000, 39.6],
-];
-
 // props type declaration
 export type DetailsProps = {
+  token: string;
+  id: string;
   // function to return to the dashboard
   back: () => void;
 };
 
 // functional component that takes the name of the portfolio and a function to switch back to the dashboard
 // returns the entire details page
-const Details: React.FC<DetailsProps> = ({ back }) => {
+const Details: React.FC<DetailsProps> = ({ token, id, back }) => {
+  const [portfolioDetails, setPortfolioDetails] = React.useState<
+    API.PortfolioDetails | undefined
+  >();
+  const [error, setError] = React.useState<ErrorCode | undefined>(undefined);
+
+  const isMounted = React.useRef(true);
+
+  const fetch = async () => {
+    setError(undefined);
+    try {
+      const details = await API.details(token, id);
+      if (isMounted.current) {
+        setPortfolioDetails(details);
+      }
+    } catch (e) {
+      if (isMounted.current) {
```

Comment:
Why are we checking for isMounted here? What could go wrong if we left that check out?

Reply:
If the connection is slow, the user could navigate away from the details page, causing the `Details` component to unmount before the async fetch call is done, so the component's state is modified after unmount. Trying to modify an unmounted component causes errors and should be avoided.